### PR TITLE
Added gender bias filter

### PR DESCRIPTION
**Pull request by asnota**
The filter returns "True" when the female population is underrepresented. In all other cases, the filter returns "False".  
The cases returning "False" are listed below:
1. All sentences are tagged as "neutral", which is when:
    1) they do not contain personal pronouns,
    2) there are pronouns of both genders in the same sentence.
2. The amont of sentences tagged as "feminine" and "masculine" is equal.
3. The amont of sentences tagged as "masculine" is less than "feminine".

### CODE REVIEW

**Comment by AbinayaM02** [comment]
Hi @asnota. Thanks for the contribution. The build is failing because you have named the file as gender_bias_filter.py. The norm is to keep the file name filter.py and the folder should be named as gender_bias (to keep up with the current nomenclature). Could you please change the file name to filter.py and the folder name to gender_bias?

**Comment by asnota** [comment]
Hi @AbinayaM02 I've just made requested changes - thank you for explaining the reason of te build faluire!

File: filters/gender_bias/filter.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                feminine_flag = False
+                masculin_flag = False
+                neutral_flag = True
+                sentence_object = {
+                    "sentence": sentence,
+                    "feminine_flag": feminine_flag,
+                    "masculin_flag": masculin_flag,
+                    "neutral_flag": neutral_flag,
+                }
+            flagged_sentences.append(sentence_object)
```
Comment (AbinayaM02) [inline filters/gender_bias/filter.py:50]: The variable sentence_object needs to be initialized beforehand. Currently, there is referenced before assignment error.

**Review by AbinayaM02 — COMMENTED** [review]
Looks like the build is failing again because of an error.

File: filters/gender_bias/filter.py (hunk 1 above)
Comment (asnota) [reply filters/gender_bias/filter.py]: I've just fixed it (the faut of identation mess).

File: filters/gender_bias/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
+## gender bias filter
+
+## What type of a filter is this?
+
+This is a quadrilingual filter (for English, French, Polish and Russian languages), which filters a text corpus to measure gender fairness in regards of a female gender representation.
+It is based on a pre-defined set words, such as personal pronouns, words defining the relation, titles and names, corresponding to the female and male genders accordingly.
+Two utility methods - flag_sentences(), count_genders() and sort_groups() give supplementary information to the boolean value returned by the main filter() method.
```
Comment (AbinayaM02) [inline filters/gender_bias/README.md:7]: These lines can be removed.

File: filters/gender_bias/README.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
+## gender bias filter
+
+## What type of a filter is this?
+
+This is a quadrilingual filter (for English, French, Polish and Russian languages), which filters a text corpus to measure gender fairness in regards of a female gender representation.
```
Comment (AbinayaM02) [inline filters/gender_bias/README.md:5]: Replace "in regards of a"  with "with respect to"

File: filters/gender_bias/README.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+The beneficial impact of the current extrinsic filter is its complete transparency and extensibility, which offers the user a great amont of control over the bias measurement parameters.
+The set of names for English was taken from https://www.verywellfamily.com/, for French - from https://www.journaldesfemmes.fr, for Polish - from https://www.edziecko.pl and https://www.klups.pl, and for Russian - from https://imenika.ru/.
```
Comment (AbinayaM02) [inline filters/gender_bias/README.md:25]: Each reference can be numbered.

File: filters/gender_bias/README.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+You can check the exact number by calling the `count_genders()` static method on the instance of the class and providing it with the extracted flags with the `flag_sentences()` static method:
+```
+flagged_sentences = f.flag_sentences(sentences, "en")
+feminine, masculine, neutral = f.count_genders(flagged_sentences)
+print("Feminine flagged sentences:", feminine)
+print("Masculine flagged sentences:", masculine)
+print("Neutral flagged sentences:", neutral)
+```
+
+which outputs:
```
Comment (AbinayaM02) [inline filters/gender_bias/README.md:49]: Output:

File: filters/gender_bias/README.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+```
+This is a feminine group: ['She is going to program', 'Wow , it works!']
+This is a masculine group: ['He is going to make a cake.', 'He agreed to help him', 'Yey !!!!']
+This is a neutral group: ['Nobody likes washing dishes']
+```
+
+## What are the limitations of this filter?
+The filter result is based on n-gram intersection counting approach, which assumes that the word should have the exact form as the internally defined keywords.
+For the lexically rich languages, such as Polish and Russian, the nouns conjugation may overlap the infinitif of an opposite gender.
```
Comment (AbinayaM02) [inline filters/gender_bias/README.md:131]: Please correct the spelling mistake.

**Review by AbinayaM02 — COMMENTED** [review]
Hi @asnota : Thank you for your contribution. I have provided some minor changes in the Readme file. Please have a look.

File: filters/gender_bias/README.md (hunk 4 above)
Comment (asnota) [reply filters/gender_bias/README.md]: Dear @AbinayaM02 , thank you for your remarks - I've just submitted an updated version on Readme file accordingly.

File: filters/gender_bias/README.md (hunk 3 above)
Comment (asnota) [reply filters/gender_bias/README.md]: Updated

**Review by AbinayaM02 — APPROVED** [review]
Thank you @asnota for making the changes. 

1. The README contains all the necessary inputs about the filter.
2. test.json is present for all the four language the filter is implemented for (and it can be extended to other languages later).

Approving the PR.

**Comment by AbinayaM02** [comment]
Hi @asnota: Looks like there are conflicts in your branch. Could you please pull the main branch and fix the conflicts and push again. Thanks!

**Comment by asnota** [comment]
Hi @AbinayaM02 I pulled the main branch and pushed the changes, which are open in a pull request #285

**Comment by AbinayaM02** [comment]
Closing this PR since it is duplicated in #286 and approved.

**Pull request closed without merging**